tools/@aws-cdk/eslint-plugin/lib/rules/no-invalid-path.ts

```diff
@@ -0,0 +1,79 @@
+import { Rule } from 'eslint';
+import { isProdFile } from '../private/is-prod-file';
+import * as path from 'path';
+import * as fs from 'fs';
+
+function isPathJoinFuncCall(node: any): boolean {
+  return (
+    node.callee &&
+    node.callee.property &&
+    node.callee.property.name === 'join' &&
+    node.arguments.length > 1 &&
+    node.arguments[0].name &&
+    node.arguments[0].name === '__dirname'
+  )
+}
+
+export function create(context: Rule.RuleContext): Rule.NodeListener {
+  return {
+    CallExpression(node: any) {
+      if (!isProdFile(context.getFilename())) {
+        return;
+      }
+
+      if (isPathJoinFuncCall(node)) {
+        // Confirm path does not have unnecessary '..'
+        const pathToParentDir = '..';
+        const paths: string[] = [];
+        let forward = false; // Determines if the path is officially going forward
+        let validPath = true;
+        for (let i=1; i<node.arguments.length; i++) {
+          const path = node.arguments[i].value;
+          // If path is going forward, we cannot have '..' anymore
+          if (forward && path === pathToParentDir) {
+            validPath = false;
+            // I could break here but then I can't build the simulated path for the error message,
+            // and in practice this should be a trivial amount of extra work.
+          }
+          if (path !== pathToParentDir) {
+            forward = true;
+          }
+          paths.push(node.arguments[i].value);
```

Comment:
any reason not to use `paths.push(path);` here?

Reply:
nope, i forgot to replace it

Reply:
good catch